dns/draft-ct-over-dns.md

```diff
@@ -0,0 +1,289 @@
+
+
+## Tree Query
+
+The tree query takes as input:
+
+- `domain_for_log`
+- `start_index`
+- `leaf_index`
+- `tree_size`
+
+And returns as output:
+
+- `partial_audit_path[]`
+
+Set `QNAME` to `<i> || '.' || <leaf_index> || '.' || <tree_size> || '.tree.' || <domain_for_log> || '.'`
+
+For example, given the values above we would generate the following request:
+
+               +---------------------------------------------------+
+    Header     | OPCODE=SQUERY                                     |
+               +---------------------------------------------------+
+    Question   | QNAME=0.123456.999999.tree.pilot.ct.googleapis.com|
+               | ., QCLASS=IN, QTYPE=TXT                           |
+               +---------------------------------------------------+
+    Answer     | <empty>                                           |
+               +---------------------------------------------------+
+    Authority  | <empty>                                           |
+               +---------------------------------------------------+
+    Additional | <empty>                                           |
+               +---------------------------------------------------+
+
+And receive the following response:
+
+               +---------------------------------------------------+
+    Header     | OPCODE=SQUERY, RESPONSE, AA                       |
+               +---------------------------------------------------+
+    Question   | QNAME=0.<snip>.com., QCLASS=IN, QTYPE=TXT         |
+               +---------------------------------------------------+
+    Answer     | QNAME=0.<snip>.com., 604800 IN TXT                |
+               | <character-string for BASE64(audit_path[0])>      |
+               | <character-string for BASE64(audit_path[1])>      |
+               | <character-string for BASE64(audit_path[2])>      |
+               | ...                                               |
+               +---------------------------------------------------+
+    Authority  | <empty>                                           |
+               +---------------------------------------------------+
+    Additional | <empty>                                           |
+               +---------------------------------------------------+
+
+Where the TXT RDATA returned will be one or more character-strings, each containing the BASE64 encoding of the value for next value in the `audit_path` array.  Note that the first character-string will be for the number specified in the query, and the server MAY return one or more subsequent values that correspond to the next values in the array.  Since the size of DNS responses is limited, servers MAY not return all values for `audit_path` in a single query, and if the correct number (`length`) of response is not returned, a client MUST send a subsequent query with a different starting index until all elements are accounted for.
```

Comment:
b64 encoded? Raw binary? 7 hashes of 32 bytes, once b64 encoded, that would be 299 characters, no?